File: banda/SergioDeIscar-banda-java/SergioDeIscar-banda-java/src/main/java/org/example/models/Musician.java
package org.example.models;

public abstract class Musician extends Persona{
    protected float salario = 1500.0f;
    protected int experiencia;
    protected Musician(String nombre, int experiencia) {
        super(nombre);
        this.experiencia = experiencia;
    }

    public abstract float getSalario();

    void interpretar() {
        System.out.println("El musico " + nombre + " esta interpretando");
    }

    @Override
    public void Respirar() {
        System.out.println("El musico " + nombre + " esta respirando");
    }

    @Override
    public String getNombre() {
        return this.nombre;
    }
}

File: banda/SergioDeIscar-banda-kotlin/SergioDeIscar-banda-kotlin/src/main/kotlin/models/Musician.kt
package bandamusical.models

abstract class Musician(nombre:String): Persona(nombre) {
    open val salario: Float = 1500.0f
    abstract val experiencia: Int
    fun interpretar() {
        println("El músico $nombre interpreter")
    }
    override fun respirar() {
        println("El músico $nombre respira.")
    }
}